### PR TITLE
Delete topic

### PR DESCRIPTION
**Pull request by ouchadam**
Addresses #2 

Adds the ability to delete topics from the `admin` site

screenshot from 2018-02-23 18-02-37

Deleting a topic causes a the server to chain delete all topic references.

### CODE REVIEW

File: server/src/index.js [hunk 1]
```diff
@@ -284,16 +284,40 @@ exports.onPluginInstanceRemovedFromTopic = functions.database.ref('/v2/topics/{t
...
 })
 
+exports.onTopicDeleted = functions.database.ref('v2/topics/{topicId}').onDelete(event => {
+    const database = event.data.ref.root
+    const topicId = event.params.topicId
+
+    const removeTopicIndex = database
+        .child(`/v2/topics_index/${topicId}`)
+        .remove()
+        .catch(console.err)
```
Comment (ouchadam) [inline server/src/index.js:306]: these promises can fail if the topic has already been deleted/we're out of sync. We should log the error but allow all the other promises to continue.

File: site/admin/src/home/home-use-case.js [hunk 2]
```diff
@@ -28,6 +28,11 @@ export const deleteDevice = (dispatch) => async (device) => {
     await fetchDevices(dispatch)()
 }
 
+export const deleteTopic = (dispatch) => async (topic) => {
+    await removeTopic(topic.id)
+    await fetchTopics(dispatch)()
```
Comment (ouchadam) [inline site/admin/src/home/home-use-case.js]: we don't currently watch any database paths, so we're manually refreshing the topics after deleting

File: server/src/index.js [hunk 3]
```diff
@@ -284,16 +284,40 @@ exports.onPluginInstanceRemovedFromTopic = functions.database.ref('/v2/topics/{t
...
+
+    const removeTopicToDevices = database
+        .child(`/v2/topic_to_devices/${topicId}`)
```
Comment (takecare) [inline server/src/index.js:309]: what's `topic_to_devices`?

**Review by takecare — APPROVED** [review]

File: server/src/index.js (hunk 3 above)
Comment (ouchadam) [reply server/src/index.js:309]: it's topics indexed by deviceId. We're using firebase realtime database so we gotta build queries into the schema

**Review by lgvalle — APPROVED** [review]

**Pull request merged**